File: 917-boats-to-save-people/boats-to-save-people.py
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        people.sort()
        n =len(people)
        i,j= 0 ,n-1 
        cnt  =0
        while(i<=j):

            if people[i]+people[j]>limit:
 
                j-=1
            else:
   
                i+=1
                j-=1
            cnt+=1

            
            
            
        return cnt
            
                
            
        

File: 917-boats-to-save-people/boats-to-save-people.java


class Solution {
    public int numRescueBoats(int[] people, int limit) {
        Arrays.sort(people);
        int n = people.length;
        int i = 0, j = n - 1;
        int cnt = 0;
        while (i <= j) {
            if (people[i] + people[j] > limit) {
                j--;
            } else {
                i++;
                j--;
            }
            cnt++;
        }
        return cnt;
    }
}